modules/odometry_and_time.py

```diff
@@ -12,14 +12,27 @@ class OdometryAndTime:
     """
     Contains odometry/telemetry and timestamp.
     """
-    def __init__(self,
-                 position: drone_odometry.DronePosition,
-                 orientation: drone_odometry.DroneOrientation):
+    __create_key = object()
+
+    @classmethod
+    def create(cls, odometry_data: drone_odometry.DroneOdometry) -> "tuple[bool, OdometryAndTime | None]":
+        """
+        Private constructor, use create() method.
+        odometry_data: Mavlink/Dronekit odometry data from drone_odometry class.
+        """
+        
+        return True, OdometryAndTime(cls.__create_key, odometry_data)
+
+    def __init__(self, class_private_create_key, odometry_data: drone_odometry.DroneOdometry):
         """
         Constructor sets timestamp to current time.
         """
-        self.position = position
-        self.orientation = orientation
+        if odometry_data is None:
+            return False, None
+        
+        assert class_private_create_key is OdometryAndTime.__create_key, "Use create() method"
+        
+        self.odometry_data = odometry_data
         self.timestamp = time.time()
```

Comment:
Move this to `create()` method as well, and pass the timestamp in as a float to the constructor.